packages/taco/src/conditions/predefined/address-allowlist.ts

```diff
@@ -0,0 +1,42 @@
+import { USER_ADDRESS_PARAM_DEFAULT } from '@nucypher/taco-auth';
+
+import {
+  ContextVariableCondition,
+  ContextVariableConditionProps,
+  ContextVariableConditionType,
+} from '../base/context-variable';
+import { Condition, ERR_INVALID_CONDITION } from '../condition';
+import {
+  AddressAllowlistConditionProps,
+  addressAllowlistConditionSchema,
+} from '../schemas/address-allowlist';
+
+export { AddressAllowlistConditionProps } from '../schemas/address-allowlist';
+
+/**
+ * A Client-side condition that checks if a user's address is in an allowlist and transforms the object into `ContextVariableCondition`.
+ * The nodes process this as a standard context variable that checks if the user's address is in the allowlist.
+ * @remark This condition does not have a conditionType field and is not possible to be composed to/from JSON except as the underlying type `ContextVariableCondition`.
+ */
+export class AddressAllowlistCondition extends ContextVariableCondition {
+  constructor(value: AddressAllowlistConditionProps) {
+    const { data, error } = Condition.validate(
```

Comment:
nitpick here. I would use `AddressAllowlistCondition`. `validate` is a static function available as part of all subclasses. You'll also not need to import `Condition`.
```suggestion
    const { data, error } = AddressAllowlistCondition.validate(
```